src/js/node/events.js

```diff
@@ -356,7 +357,7 @@ function once(emitter, type, options) {
   });
 }
 
-async function* on(emitter, event, options = {}) {
+function on(emitter, event, options = {}) {
```

Comment:
Needed to change this to a normal function, else events were not stored in the queue before the first `.next()` call.